cmd/all-in-one/main.go

```diff
@@ -279,7 +280,10 @@ func startQuery(
 ) *queryApp.Server {
 	spanReader = storageMetrics.NewReadMetricsDecorator(spanReader, metricsFactory)
 	qs := querysvc.NewQueryService(spanReader, depReader, *queryOpts)
-	server, err := queryApp.NewServer(svc.Logger, svc.HC(), qs, metricsQueryService, qOpts, tm, jt)
+	server, err := queryApp.NewServer(telemetery.TelemeterySetting{
```

Comment:
nit: my preference is to always pass `telset` as the last argument, because it's a de-facto dependency, not the business functionality that should be at the front of arg list.

The healthcheck should also be part of the `telset`, see how it's done in OTEL.

Lastly, why is this the first change? I would expect this function to already receive a `telset` param, not create a new one. `telset` should only be created once in the main function.

Reply:
The initialization of the Telemetery Struct can't be done from the source itself, because different components have different requirements. Some mains() are passing jTracer.NoOp, whereas some aren't passing any Tracer/Metrics Factory itself.
So, the telemetery struct will have different initialization every time.
Am I right?
As for the main function, yes I will create telset param only once.

Reply:
different mains can have different initialization, although it would be good to understand which ones are doing jTracer.NoOp and why